src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileParser.cs

```diff
@@ -14,26 +16,34 @@
 using Uno.Roslyn;
 using Windows.Foundation.Metadata;
 using System.Threading;
-using System.IO;
+using Microsoft.CodeAnalysis.Text;
 
 #if NETFRAMEWORK
 using GeneratorExecutionContext = Uno.SourceGeneration.GeneratorExecutionContext;
 #endif
 
 namespace Uno.UI.SourceGenerators.XamlGenerator
 {
-	internal class XamlFileParser
+	internal partial class XamlFileParser
 	{
+		private static readonly ConcurrentDictionary<CachedFileKey, CachedFile> _cachedFiles = new();
+		private static readonly TimeSpan _cacheEntryLifetime = new TimeSpan(hours: 1, minutes: 0, seconds: 0);
+		private readonly string _excludeXamlNamespacesProperty;
+		private readonly string _includeXamlNamespacesProperty;
 		private readonly string[] _excludeXamlNamespaces;
 		private readonly string[] _includeXamlNamespaces;
 		private readonly RoslynMetadataHelper _metadataHelper;
 		private int _depth;
 
-		public XamlFileParser(string[] excludeXamlNamespaces, string[] includeXamlNamespaces, RoslynMetadataHelper roslynMetadataHelper)
+		public XamlFileParser(string excludeXamlNamespaces, string includeXamlNamespaces, RoslynMetadataHelper roslynMetadataHelper)
 		{
-			_excludeXamlNamespaces = excludeXamlNamespaces;
-			_includeXamlNamespaces = includeXamlNamespaces;
-			this._metadataHelper = roslynMetadataHelper;
+			_excludeXamlNamespacesProperty = excludeXamlNamespaces;
+			_excludeXamlNamespaces = excludeXamlNamespaces.Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
+
+			_includeXamlNamespacesProperty = includeXamlNamespaces;
+			_includeXamlNamespaces = includeXamlNamespaces.Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
+			
+			_metadataHelper = roslynMetadataHelper;
```

Comment:
@jeromelaban `_metadataHelper` is used in parsing for conditional namespaces, for example:

https://github.com/unoplatform/uno/blob/88d7a4024ccdcaa07a6b8cd056ba55ef4e92865e/src/Uno.UI.Tests/Windows_UI_Xaml/Controls/When_Conditional_IsTypePresent_Page.xaml#L5

It's not clear to me if it's possible to have a wrong cache entry due to that?

Reply:
Good catch! It theoretically could, particularly if the cache is reused between versions of Uno. This feature only works on windows when used with types coming from UWP/WinUI itself, but not for other types. One way to fix this would be to include the information version number of the `Uno.UI` assembly (when building outside of Uno.UI itself) and use it as a cache key as well.

Reply:
@jeromelaban Is `Uno.UI` version the only factor here? I thought things like conditional directives, etc, could affect the outcome of `IsTypePresent` as well?

One idea is to not cache files using `IsTypePresent`, I assume most files don't use it so we'll still do a good amount of caching. Can you confirm if this approach sounds good?